### ISSUE TITLE
Docker image + suggestion

### ISSUE THREAD

**Issue by PierreDurrr** [CollinHeist/TPDbCollectionMaker#3]
Not really an issue as it's already a great tool.
But a suggestion : maybe a docker image, plus some sort of watchdog service.
i.e.: copy your .html to some predefined folder and boom, it serve you a ready to use .yml file, maybe keeping original filename.
I'm already trying to mess with it myself, but i'm far from something working...

**Comment by CollinHeist** [comment]
@PierreDurrr Hey, thanks for the suggestion. That's a cool idea. I don't think I'll have time to implement it any time soon, but I welcome a PR & I'll keep the issue open for if/when I get the time.

**Comment by PierreDurrr** [comment]
@CollinHeist There you go : https://github.com/PierreDurrr/TPDBCollectionMaker-docker
**Still a work in progress** as watchdog is partially working (no output file yet).
Also not keeping original filename for the moment.

Code is probably a bit messy as i'm not a docker professional...

**Comment by PierreDurrr** [comment]
Aaaaand there you : [https://github.com/PierreDurrr/TPDBCollectionMaker-docker/tree/webui](https://github.com/PierreDurrr/TPDBCollectionMaker-docker/tree/webui) (webui branch)
Flask webui tied to watchdog and your script.
Feel free to review, test, comment, fork, whatever... if you want !

**Comment by CollinHeist** [comment]
@PierreDurrr Looks great! Very nice work. Did you want it merged back into this project, for wider use?

**Comment by PierreDurrr** [comment]
Haha thanks !
Yes, can be merged if you want. Keep in mind that's probably still a bit messy, but it's working 😅

**Comment by CollinHeist** [comment]
If you want to submit a PR I'll make some edits and merge that when I have time. I think the "enter URL" functionality will have to be removed, as that's a violation of TPDb's TOS (they're pretty picky)

**Comment by Investigamer** [comment]
Just stumbled here while working on a similar project, some neat stuff! Dockerized would definitely be ideal, I might be able to help with that. 

Sidenote: It's interesting that the README says "TPDb doesn't permit automated scraping", are we talking some kind of ToS or a technical obstacle? 

Because I've found it to be pretty easy to scrape responsibly lol. If a large source of publicly listed data is repeatedly asked nicely to provide some kind of basic common courtesy public API, and they don't... I mean, they can't really blame you for trying to grab that data the hard way. 

They can certainly rate limit you or block you if you make the requests irresponsibly fast and numerous, but there's certainly no law against it. At the end of the day, there's no difference between browsing the site yourself, and having a bot browse it, as long as the rate at which the bot browses it is within reason (ratelimit package lets you control this gracefully). I'm not sure how visiting each page and downloading it as HTML beforehand is any better than having the requests package do it?

**Comment by CollinHeist** [comment]
@Investigamer 

> are we talking some kind of ToS or a technical obstacle?

It is against their site TOS, and they have made strongly worded... "suggestions"... to me (regarding this tool) in the past, before I explained it does not perform any automated actions.

> Because I've found it to be pretty easy to scrape responsibly lol. If a large source of publicly listed data is repeatedly asked nicely to provide some kind of basic common courtesy public API, and they don't... I mean, they can't really blame you for trying to grab that data the hard way.

Although I agree _in spirit_, they do have a right to control how users interact with their site and content (including banning users of automatic scrapers, etc.) 

> there's no difference between browsing the site yourself, and having a bot browse it, [...]

Well not quite, one of these is not a violation of their TOS, the other is 😉 

---

Overall I think TPDb _should_ develop an API (or permit limited scraping until they do..), but I'm not personally interested in maintaining software which pushes the boundaries of a free-service in that way

**Comment by Investigamer** [comment]
Ok gotcha, thanks for the quick response buddy! Was just curious about the reason and your thinking on the matter, that's totally understandable 👍

**Comment by CollinHeist** [comment]
Yeah no problem. I *really* wish they just had an API, then there wouldn't be a need for weird workarounds like this, haha. But for now, it works 🤷🏻
